### PR TITLE
Urbdrc fixes

### PR DESCRIPTION
**Pull request by akallabeth**
- Replaces pthread_mutex with winpr functions
- Replaces pthread with winpr functions
- Removes `zfree` to increas performance.
- Adds some error checking (lots still to be done)

### CODE REVIEW

File: channels/urbdrc/client/isoch_queue.c [hunk 1]
```diff
@@ -69,18 +70,24 @@ static ISOCH_CALLBACK_DATA* isoch_queue_register_data(ISOCH_CALLBACK_QUEUE* queu
 		queue->tail->next = (void*)isoch;
 		isoch->prev = (void*)queue->tail;
 		queue->tail = isoch;
+		isoch = NULL;
```
Comment (mfleisz) [inline channels/urbdrc/client/isoch_queue.c]: This results in incorrectly returning NULL even though the item was successfully added to the queue

File: channels/urbdrc/client/libusb/request_queue.c [hunk 2]
```diff
@@ -71,50 +74,63 @@ TRANSFER_REQUEST* request_queue_register_request(REQUEST_QUEUE* queue, UINT32 Re
 		queue->tail->next = (void*) request;
 		request->prev = (void*) queue->tail;
 		queue->tail = request;
+		request = NULL;
```
Comment (mfleisz) [inline channels/urbdrc/client/libusb/request_queue.c]: This results in incorrectly returning NULL even though the item was successfully added to the queue

**Comment by Fantu** [comment]
@akallabeth is this PR still "active"? I tried your newer branch urb_fix (after rebasing it to latest master) but don't solve my "strange" problem, is this correct updated branch with usb redirection fixes?
I tested usb redirection with freerdp 1-2 years ago successfull except frequest crashes.
Now I retried it latest nightly build and is not working, even if I already tried to add udev rules for change usb device group to plugdev (where the user is) like what I did time ago but still give me this error:
```
./xfreerdp /f /bpp:32 /u:u1 /v:192.168.1.235 /dvc:urbdrc,id,dev:046d:082b
[10:41:45:582] [7721:7722] [INFO][com.freerdp.client.common.cmdline] - loading channelEx drdynvc
...
[10:41:51:631] [7721:7722] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel urbdrc
[10:41:51:631] [7721:7722] [ERROR][com.freerdp.channels.urbdrc.client] - error processing arguments
[10:41:51:646] [7721:7722] [WARN][com.freerdp.channels.urbdrc.client] - bus:0 dev:0 not exist in udevman
[10:41:51:646] [7721:7722] [INFO][com.freerdp.channels.urbdrc.client] - VID: 0x046D, PID: 0x082B
[10:41:51:724] [7721:7722] [WARN][com.freerdp.channels.urbdrc.client] - bus:1 dev:7 not exist in udevman
[10:41:52:124] [7721:7730] [ERROR][com.freerdp.channels.urbdrc.client] - unknown FunctionId 0x1
....
```
the udev rules seems working:
`ls -alh /dev/bus/usb/001/
total 0
drwxr-xr-x 2 root root       160 Feb 16 09:00 .
drwxr-xr-x 3 root root        60 Jan  1  1970 ..
crw-rw-r-- 1 root plugdev 189, 0 Feb 15 15:05 001
crw-rw-r-- 1 root plugdev 189, 1 Feb 15 15:05 002
crw-rw-r-- 1 root plugdev 189, 2 Feb 15 15:05 003
crw-rw-r-- 1 root plugdev 189, 3 Feb 15 15:05 004
crw-rw-r-- 1 root plugdev 189, 4 Feb 15 15:05 005
crw-rw-r-- 1 root plugdev 189, 6 Feb 16 09:15 007
`
I also tried 666 permission to devices and also run xfreerdp as sudo, in all cases give the same error.

Can someone tell me what I did wrong please? Or it is a recent regression?
Thanks for any reply and sorry for my bad english.

**Comment by akallabeth** [comment]
@Fantu Sorry, did not have enough time to complete that one. Closing for now.
As for your issue, looks like there is some issue with the arguments you use. 
Redirection is working on master (just tested a webcam on debian), although the code still needs work.

**Pull request closed without merging**